pkg/querier/queryrange/queryrange.proto

```diff
@@ -33,7 +33,7 @@ message LokiRequest {
   logproto.Direction direction = 6;
   string path = 7;
   repeated string shards = 8 [(gogoproto.jsontag) = "shards"];
-  bytes plan = 10 [(gogoproto.customtype) = "github.com/grafana/loki/pkg/querier/plan.QueryPlan"];
+  Plan plan = 10 [(gogoproto.customtype) = "github.com/grafana/loki/pkg/querier/plan.QueryPlan"];
```

Comment:
why do we want to use `gogoproto.customtype` here instead of defining a type in proto?

Reply:
> why do we want to use gogoproto.customtype here instead of defining a type in proto?

So far the AST is serialized as JSON. So this was faster. The plan should be defined in Protobuf eventally. Especially when it starts to differ from the AST but this is for future work 🙂